presto-main/src/main/java/io/prestosql/operator/unnest/Unnester.java

```diff
@@ -70,10 +70,10 @@ public final void startNewOutput(PageBuilderStatus status, int expectedEntries)
         }
     }
 
-    public final void processCurrentAndAdvance(int requiredOutputCount)
+    public final void processCurrentAndAdvance(int requiredOutputCount, boolean doubleRows)
     {
         checkState(currentPosition >= 0 && currentPosition < getInputEntryCount(), "position out of bounds");
-        processCurrentPosition(requiredOutputCount);
+        processCurrentPosition(requiredOutputCount, doubleRows);
```

Comment:
It would change the order in which the rows are produced, so it would require a change in the replicate block builder and in the way ordinality and markers are appended.
But I think the code would be easier to read if I changed it as you suggest.